tests/unit/checkmate/async/tasks_test.py

```diff
@@ -0,0 +1,58 @@
+from contextlib import contextmanager
+from unittest.mock import sentinel
+
+import pytest
+from requests import RequestException
+
+from checkmate.app import configure
+from checkmate.async.tasks import sync_blocklist
+
+
+class TestSyncBlocklist:
+    def test_it_works_with_url(self, CustomRules, pyramid_request):
+        pyramid_request.registry.settings["checkmate_blocklist_url"] = sentinel.url
+
+        sync_blocklist()
+
+        CustomRules.assert_called_once_with(pyramid_request.db)
+        CustomRules.return_value.load_simple_rule_url.assert_called_once_with(
+            sentinel.url
+        )
+
+    def test_it_does_nothing_without_url(self, CustomRules, pyramid_request):
+        pyramid_request.registry.settings["checkmate_blocklist_url"] = None
+
+        sync_blocklist()
+
+        CustomRules.assert_not_called()
+
+    def test_it_handles_request_exceptions(self, CustomRules, pyramid_request):
+        pyramid_request.registry.settings["checkmate_blocklist_url"] = sentinel.url
```

Comment:
This repeats the above, but I was thinking about our "not making things too complicated" idea. Might be clearer to explicitly list it unless we get hundreds requiring it

Reply:
Yeah, I think it's probably better test code to have each test's first line set the appropriate `pyramid_settings["checkmate_blocklist_url"]` for that test. I think you can use the `pyramid_settings` fixture though, instead of saying `pyramid_request.registry.settings` every time?

Reply:
You can, but this reflects how the code accesses it. `pyramid_settings` totally works, but why it works is a little more involved.